### DISCUSSION TITLE
SQL - Fill values in fields based on conditions

### DISCUSSION THREAD

**Discussion by itsmichaelhagen** [community/community#131067] [Programming Help] [answered]
### Body

Hello,

I'm new to SQL and doing a little practice. I want to do a "VLOOKUP" type of update on this table, but I'm unsure how.

I have 2 tables, one that contains "teams" and one that contains "players". It is a database of professional baseball players, so when adding minor league players there will be different levels. The idea was to designate what organization they play for (i.e. the Yankees), then to say what level they play at ("AAA", "AA", "MLB", etc.). Then, I would like to fill in their _**team**_ based on the level and organization they play at. For example, Seth Beer (awesome name) plays for the _**Pittsburgh**_ organization in _**AAA**_. That information would be provided in the _**players**_ table with the entire list of teams, their levels, and what organization they are affiliated with in the _**teams**_ table. I would like to setup a function or something where it would automatically fill in, based on my _**teams*_* table, that his current team is _**Indianapolis**_ so that as this database gets bigger, I don't have to continue tracking down the names of the hundreds of minor league teams to type in manually every update.

If anyone can help me figure this out, I would greatly appreciate it!

-Rookie SQL-er

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by davevad93** [comment, marked as answer]
Hi @itsmichaelhagen ,

To automate this process you can create a trigger that updates the team every time a new player is inserted or updated. I don't know the exact structure of your DB, so the following example is just a guide.

Let's assume you have these two tables:

```sql
teams (
    team_id INT PRIMARY KEY,
    team_name VARCHAR(100),
    organization VARCHAR(100),
    level VARCHAR(10)
);
``` 

```sql
players (
    player_id INT PRIMARY KEY,
    player_name VARCHAR(100),
    organization VARCHAR(100),
    level VARCHAR(10),
    team_name VARCHAR(100) 
);
``` 

This would be the trigger to update the `team_name` field that will be executed after an `INSERT` or `UPDATE` on the `players` table:

```sql
CREATE TRIGGER update_team_name
AFTER INSERT OR UPDATE ON players
FOR EACH ROW
BEGIN
    -- Update the 'players' table, setting the 'team_name' column
    UPDATE players p
    SET p.team_name = t.team_name
    -- Specify the 'teams' table as 't' for the join operation
    FROM teams t
    -- Join condition to match the 'organization' and 'level' in the 'players' table with those in the 'teams' table
    WHERE p.organization = t.organization
      AND p.level = t.level
      -- Ensure that only the newly inserted or updated row in the 'players' table is updated
      AND p.player_id = NEW.player_id;
END;
```

As I said, this is just an example, you should adjust the trigger according to the structure and type of your DB.

Here are some useful resources about SQL triggers:

1. [The Comprehensive Guide to SQL Triggers - SQL Tutorial](https://www.sqltutorial.org/sql-triggers/)
2. [Learn SQL: SQL Triggers](https://www.sqlshack.com/learn-sql-sql-triggers/)
3. [SQL Triggers: What They Are and How to Use Them](https://www.dbvis.com/thetable/sql-triggers-what-they-are-and-how-to-use-them/)

Hope this helps!

**Reply by itsmichaelhagen** [reply]
This is exactly what I was looking for, thanks man!

**Reply by davevad93** [reply]
You're welcome man, a couple of years ago I took an exam where I had to do something similar to this but with NBA players and teams.
